packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -200,6 +201,52 @@ export class Controller<
         }
     }
 
+    /**
+     * Adds the behavior to the component.
+     * @param behavior - The behavior to add.
+     */
+    addBehavior(behavior: HostBehavior<TElement>) {
+        const targetBehaviors = this.behaviors ?? (this.behaviors = new Map());
+        const count = targetBehaviors.get(behavior) ?? 0;
+
+        if (count === 0) {
+            targetBehaviors.set(behavior, 1);
+            behavior.addedCallback && behavior.addedCallback(this);
+
+            if (behavior.connectedCallback && this.isConnected) {
+                behavior.connectedCallback(this);
+            }
+        } else {
+            targetBehaviors.set(behavior, count + 1);
+        }
+    }
+
+    /**
+     * Removes the behavior from the component.
+     * @param behavior - The behavior to remove.
+     * @param force - Forces removal even if this behavior was added more than once.
+     */
+    removeBehavior(behavior: HostBehavior<TElement>, force: boolean = false) {
+        const targetBehaviors = this.behaviors;
+        if (targetBehaviors === null) {
+            return;
+        }
+
+        const count = (targetBehaviors.get(behavior) ?? 0) - 1;
```

Comment:
It seems `count` can be `-1` here, which causes execution of the else block, setting the behavior key to - 1. Is that intentional? I think you could enter that state by calling `removeBehavior` with a behavior that was never added.